server/api/db/sqldb/db.py

```diff
@@ -4180,6 +4183,182 @@ def create_data_version(self, session, version):
+            query = query.filter(AlertConfig.project == project)
+
+        alerts = list(map(self._transform_alert_config_record_to_schema, query.all()))
+        for alert in alerts:
+            self.enrich_alert(session, alert)
+        return alerts
+
+    def get_alert(
+        self, session, project: str, name: str
+    ) -> mlrun.common.schemas.AlertConfig:
+        return self._transform_alert_config_record_to_schema(
+            self._get_alert(session, name, project)
+        )
+
+    def get_alert_by_id(
+        self, session, alert_id: int
+    ) -> mlrun.common.schemas.AlertConfig:
+        resp = self._query(session, AlertConfig, id=alert_id).one_or_none()
+        return self._transform_alert_config_record_to_schema(resp)
+
+    def enrich_alert(self, session, alert: AlertConfig):
+        state = self.get_alert_state(session, alert.id)
+        alert.state = (
+            mlrun.common.schemas.AlertActiveState.ACTIVE
+            if state.active
+            else mlrun.common.schemas.AlertActiveState.INACTIVE
+        )
+        alert.count = state.count
+
+        def _enrich_notification(_notification):
+            _notification = _notification.to_dict()
+            # we don't want to return the secret
+            del _notification["secret_params"]
+
+            if not isinstance(_notification["when"], list):
+                _notification["when"] = [_notification["when"]]
+            return _notification
+
+        alert.notifications = [
+            mlrun.common.schemas.notification.Notification(
+                **_enrich_notification(notification)
+            )
+            for notification in self._get_db_notifications(
+                session, AlertConfig, parent_id=alert.id
+            )
+        ]
+
+    @staticmethod
+    def _transform_alert_config_record_to_schema(
+        alert_config_record: AlertConfig,
+    ) -> mlrun.common.schemas.AlertConfig:
+        if alert_config_record is None:
+            return None
+
+        alert = mlrun.common.schemas.AlertConfig(**alert_config_record.full_object)
+        alert.id = alert_config_record.id
+        return alert
+
+    @staticmethod
+    def _transform_alert_config_schema_to_record(
+        alert: mlrun.common.schemas.AlertConfig,
+    ) -> AlertConfig:
+        alert_record = AlertConfig(
+            id=alert.id,
+            name=alert.name,
+            project=alert.project,
+        )
+        alert_record.full_object = alert.dict()
+        return alert_record
+
+    def _get_alert(self, session, name: str, project: str) -> AlertConfig:
+        try:
+            return self._query(
+                session, AlertConfig, name=name, project=project
+            ).one_or_none()
+        finally:
+            pass
+
+    def _get_alert_by_id(self, session, alert_id: int) -> AlertConfig:
+        try:
+            return self._query(session, AlertConfig, id=alert_id).one_or_none()
+        finally:
+            pass
```

Comment:
Looks like the difference between this and the `get_alert_by_id` is that this returns the DB object, while the other method returns the schema. Some questions:
1. Why not have the other method call this one?
2. Also - why do you need the `try` and `finally` here?
3. And finally, I suggest using a different name to indicate they are returning different objects.